lightcurve/alcdef.py

```diff
@@ -1,52 +1,63 @@
-from django.contrib import admin
-from django.db import models
-from django.http import HttpResponse
+ALLOWED_KEYWORDS = [
+    'REVISEDDATA',
+    'OBJECTNUMBER',
+    'OBJECTNAME',
+    'MPCDESIG',
+    'CONTACTNAME',
+    'CONTACTINFO',
+    'OBSERVERS',
+    'OBSLONGITUDE',
+    'OBSLATITUDE',
+    'SESSIONDATE',
+    'SESSIONTIME',
+    'FILTER',
+    'MAGBAND',
+    'STANDARD',
+    'DIFFERMAGS',
+    'LTCTYPE',
+    'LTCDAYS',
+    'LTCAPP',
+    'REDUCEDMAGS',
+    'UCORMAG',
+    'OBJECTRA',
+    'OBJECTDEC',
+    'PHASE',
+    'PABL',
+    'PABB',
+    'CICORRECTION',
+    'CIBAND',
+    'CITARGET',
+    'PUBLICATION',
+    'BIBCODE',
+    'DELIMITER',
+    'COMMENT',
+]
 
-class AlcdefWriter(models.Model):
+REQUIRED_KEYWORDS = [
+    'CONTACTINFO',
+    'CONTACTNAME',
+    'DELIMITER',
+    'FILTER',
+    'OBJECTNAME',
+    'OBJECTNUMBER',
+    'OBSERVERS',
+    'SESSIONDATE',
+    'SESSIONTIME',
+]
+
+class AlcdefWriter(object):
     def __init__(self):
-        self.metadata = {
-            'REVISEDDATA': '',
-            'OBJECTNUMBER': '',
-            'OBJECTNAME': '',
-            'MPCDESIG': '',
-            'CONTACTNAME': '',
-            'CONTACTINFO': '',
-            'OBSERVERS': '',
-            'OBSLONGITUDE': '',
-            'OBSLATITUDE': '',
-            'SESSIONDATE': '',
-            'SESSIONTIME': '',
-            'FILTER': '',
-            'MAGBAND': '',
-            'STANDARD': '',
-            'DIFFERMAGS': '',
-            'LTCTYPE': '',
-            'LTCDAYS': '',
-            'LTCAPP': '',
-            'REDUCEDMAGS': '',
-            'UCORMAG': '',
-            'OBJECTRA': '',
-            'OBJECTDEC': '',
-            'PHASE': '',
-            'PABL': '',
-            'PABB': '',
-            'CICORRECTION': '',
-            'CIBAND': '',
-            'CITARGET': '',
-            'PUBLICATION': '',
-            'BIBCODE': '',
-            'DELIMITER': '',
-            'COMMENT': '',
-        }
+        self.metadata = {}
         self.starid = 1 # ID for metadata of comparison star
         self.comments = []
         self.datas = []
 
     def set(self, key, value):
-        self.metadata[key] = value
+        if key in ALLOWED_KEYWORDS:
+            self.metadata[key] = value
```

Comment:
This would be a good place to warn the user

```
else:
   raise KeyError('Not a valid ALCDEF key: %s' % key)
```

Reply:
Done.